src/Cli/dotnet/commands/dotnet-workload/list/WorkloadListCommand.cs

```diff
@@ -105,29 +106,23 @@ public override int Execute()
             return 0;
         }
 
-        internal UpdateAvailableEntry[] GetUpdateAvailable(IEnumerable<WorkloadId> installedList)
+        internal IEnumerable<UpdateAvailableEntry> GetUpdateAvailable(IEnumerable<WorkloadId> installedList)
         {
-            HashSet<WorkloadId> installedWorkloads = installedList.ToHashSet();
```

Comment:
How many ids are in this list? Making this change could be a perf regression if the number is large, though I'd be a bit surprised by that.

Reply:
Noah said something similar. I responded to that concern here: https://github.com/dotnet/sdk/pull/35859/files#r1347950133